pkg/build/controller/controller.go

```diff
@@ -190,6 +168,48 @@ func (bc *BuildController) nextBuildPhase(build *buildapi.Build) error {
 	return nil
 }
 
+// resolveOutputDockerImageReference returns a reference to a Docker image
+// computed from the buid.Spec.Output.To reference.
+func (bc *BuildController) resolveOutputDockerImageReference(build *buildapi.Build) (string, error) {
+	outputTo := build.Spec.Output.To
+	if outputTo == nil || outputTo.Name == "" {
+		return "", nil
+	}
+	var ref string
+	switch outputTo.Kind {
+	case "DockerImage":
+		ref = outputTo.Name
+	case "ImageStream", "ImageStreamTag":
+		// TODO(smarterclayton): security, ensure that the reference image stream is actually visible
```

Comment:
hehe ;)

I'd love to solve that one, need pointers to how to apply the visibility rules. Any clues?

Since moving around the TODO, I wanted to make it easier to remember that @smarterclayton is probably the go-to person to explain what is to be done, as he wrote that first :)
We have many other named TODOs, thought that was a common practice here.